src/Raven.Server/Documents/Patch/PatchDocument.cs

```diff
@@ -267,9 +268,17 @@ protected PatcherOperationScope ApplySingleScript(DocumentsOperationContext cont
             }
         }
 
-        protected void SetupInputs(Document document, PatcherOperationScope scope, Engine jintEngine)
+        protected void SetupInputs(Document document, PatcherOperationScope scope, Engine jintEngine, string docId)
         {
-            jintEngine.SetValue("__document_id", document.Key);
+            jintEngine.SetValue("__document_id", docId);
+
+            if (document == null)
```

Comment:
By doing this you will ensure that you do not have unwanted functionality when doing simpler patching.
Do it here maybe: https://github.com/ravendb/ravendb/blob/v4.0/src/Raven.Server/Documents/Patch/PatchDocument.cs#L117